schema/string.go

```diff
@@ -17,18 +17,24 @@ const (
 	stringUUIDVersion = 4
 )
 
-func checkStringConstraints(v string, minLength, maxLength int, t string) error {
+func checkStringConstraints(v string, minLength, maxLength int, pattern *regexp.Regexp) error {
```

Comment:
Suggestion: use `c Constraints` as a parameter and avoid adding more parameters? The method name already says check constraints.

Reply:
Agree